linode/token/framework_resource.go

```diff
@@ -0,0 +1,270 @@
+package token
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/linode/linodego"
+	"github.com/linode/terraform-provider-linode/linode/helper"
+)
+
+func NewResource() resource.Resource {
+	return &Resource{}
+}
+
+type Resource struct {
+	client *linodego.Client
+}
+
+func (data *ResourceModel) parseToken(token *linodego.Token) {
+	data.Created = types.StringValue(token.Created.Format(time.RFC3339))
+	data.Expiry = types.StringValue(token.Expiry.Format(time.RFC3339))
+	data.Label = types.StringValue(token.Label)
+	data.Scopes = types.StringValue(token.Scopes)
+	data.Token = types.StringValue(token.Token)
+	data.ID = types.StringValue(strconv.Itoa(token.ID))
+}
+
+func (r *Resource) Configure(
+	ctx context.Context,
+	req resource.ConfigureRequest,
+	resp *resource.ConfigureResponse,
+) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	meta, ok := req.ProviderData.(*helper.FrameworkProviderMeta)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf(
+				"Expected *http.Client, got: %T. Please report this issue to the provider developers.",
+				req.ProviderData,
+			),
+		)
+
+		return
+	}
+
+	r.client = meta.Client
```

Comment:
This might be useful as a helper function to be reused across all resources.

Reply:
Just wrapped the get meta logic into a helper function. Thank you for the suggestion!